sdk/data/aztables/error_codes.go

```diff
@@ -0,0 +1,107 @@
+	DuplicatePropertiesSpecified TableErrorCode = "DuplicatePropertiesSpecified"
+	EntityAlreadyExists          TableErrorCode = "EntityAlreadyExists"
+	EntityNotFound               TableErrorCode = "EntityNotFound"
+	EntityTooLarge               TableErrorCode = "EntityTooLarge"
+	HostInformationNotPresent    TableErrorCode = "HostInformationNotPresent"
+	InvalidDuplicateRow          TableErrorCode = "InvalidDuplicateRow"
+	InvalidInput                 TableErrorCode = "InvalidInput"
+	InvalidValueType             TableErrorCode = "InvalidValueType"
+	JSONFormatNotSupported       TableErrorCode = "JsonFormatNotSupported"
+	MethodNotAllowed             TableErrorCode = "MethodNotAllowed"
+	NotImplemented               TableErrorCode = "NotImplemented"
+	OutOfRangeInput              TableErrorCode = "OutOfRangeInput"
+	PropertiesNeedValue          TableErrorCode = "PropertiesNeedValue"
+	PropertyNameInvalid          TableErrorCode = "PropertyNameInvalid"
+	PropertyNameTooLong          TableErrorCode = "PropertyNameTooLong"
+	PropertyValueTooLarge        TableErrorCode = "PropertyValueTooLarge"
+	ResourceNotFound             TableErrorCode = "ResourceNotFound"
+	TableAlreadyExists           TableErrorCode = "TableAlreadyExists"
+	TableBeingDeleted            TableErrorCode = "TableBeingDeleted"
+	TableNotFound                TableErrorCode = "TableNotFound"
+	TooManyProperties            TableErrorCode = "TooManyProperties"
+	UpdateConditionNotSatisfied  TableErrorCode = "UpdateConditionNotSatisfied"
+	XMethodIncorrectCount        TableErrorCode = "XMethodIncorrectCount"
+	XMethodIncorrectValue        TableErrorCode = "XMethodIncorrectValue"
+	XMethodNotUsingPost          TableErrorCode = "XMethodNotUsingPost"
+)
+
+// PossibleTableErrorCodeValues returns a slice of all possible TableErrorCode values
+func PossibleTableErrorCodeValues() []TableErrorCode {
+	return []TableErrorCode{
+		DuplicatePropertiesSpecified,
+		EntityAlreadyExists,
+		EntityNotFound,
+		EntityTooLarge,
+		HostInformationNotPresent,
+		InvalidDuplicateRow,
+		InvalidInput,
+		InvalidValueType,
+		JSONFormatNotSupported,
+		MethodNotAllowed,
+		NotImplemented,
+		OutOfRangeInput,
+		PropertiesNeedValue,
+		PropertyNameInvalid,
+		PropertyNameTooLong,
+		PropertyValueTooLarge,
+		TableAlreadyExists,
+		TableBeingDeleted,
+		TableNotFound,
+		TooManyProperties,
+		UpdateConditionNotSatisfied,
+		XMethodIncorrectCount,
+		XMethodIncorrectValue,
+		XMethodNotUsingPost,
+	}
+}
+
+type oDataError struct {
+	Error oDataMessage `json:"odata.error"`
+}
+
+type oDataMessage struct {
+	Code TableErrorCode `json:"code"`
+}
+
+// parseErrorCode creates a azcore.ResponseError with the Code as one of the TableErrorCodes
+func parseErrorCode(svcErr error) error {
+	var httpErr *azcore.ResponseError
+	if errors.As(svcErr, &httpErr) {
+		body, err := ioutil.ReadAll(httpErr.RawResponse.Body)
+		if err != nil {
+			return svcErr
+		}
+
+		var oerr oDataError
+		err = json.Unmarshal(body, &oerr)
```

Comment:
I'm surprised this is JSON.  When I was testing the changes for `SetAccessPolicy` the error response was XML.
```xml
<?xml version="1.0" encoding="utf-8"?>
<m:error xmlns:m="http://schemas.microsoft.com/ado/2007/08/dataservices/metadata">
  <m:code>InvalidXmlDocument</m:code>
  <m:message xml:lang="en-US">
    XML specified is not syntactically valid.
    RequestId:84770ad1-9002-00ae-1d5d-4fc246000000
    Time:2022-04-13T17:39:33.4024706Z</m:message>
</m:error>
```
Do you know when the error response is JSON vs XML?

Reply:
I assume Sean tested this, right?

Reply:
Also, I submitted a fix to handle the XML case when namespaces are involved.  Once that fix is in, the XML case should just work.